caffe2/operators/reduction_ops.h

```diff
@@ -14,12 +14,16 @@ class SumElementsOp : public Operator<Context> {
  public:
   USE_OPERATOR_CONTEXT_FUNCTIONS;
 
-  template <class... Args>
-  explicit SumElementsOp(Args&&... args)
-      : Operator<Context>(std::forward<Args>(args)...),
+  explicit SumElementsOp(const OperatorDef& operator_def, Workspace* ws)
+      : Operator<Context>(operator_def, ws),
         average_(this->template GetSingleArgument<bool>("average", false)) {}
   explicit SumElementsOp(const OperatorDef& operator_def, Workspace* ws, bool average)
       : Operator<Context>(operator_def, ws), average_(average) {}
+  explicit SumElementsOp(const c10::FunctionSchema& schema, std::vector<c10::IValue> inputs, std::vector<c10::IValue*> outputs)
+      : Operator<Context>(schema, std::move(inputs), std::move(outputs)),
+        average_(this->template GetSingleArgument<bool>("average", false)) {}
+  explicit SumElementsOp(const c10::FunctionSchema& schema, std::vector<c10::IValue> inputs, std::vector<c10::IValue*> outputs, bool average)
+      : Operator<Context>(schema, std::move(inputs), std::move(outputs)), average_(average) {}
```

Comment:
What's going on here?

Reply:
SumElementsOp() is both, an operator and a base class for other operators. When used as an operator, it is constructed with `(operator_def, workspace)`, when used as a base class it is constructed with `(operator_def, workspace, average)`. So unfortunately, one constructor that just passes through all arguments to the `Operator` constructor doesn't work here and I need to spell out all variants.